packages/rum-core/src/domain/rumEventsCollection/resource/matchResponseToPerformanceEntry.ts

```diff
@@ -0,0 +1,110 @@
+import type { RelativeTime, Duration, ClocksState } from '@datadog/browser-core'
+import { addDuration, elapsed, timeStampNow, addTelemetryDebug } from '@datadog/browser-core'
+import type { RequestCompleteEvent } from '../../requestCollection'
+import { toValidEntry } from './resourceUtils'
+
+export interface RumPerformanceResourceTiming {
+  entryType: 'resource'
+  initiatorType: string
+  name: string
+  startTime: RelativeTime
+  duration: Duration
+  fetchStart: RelativeTime
+  domainLookupStart: RelativeTime
+  domainLookupEnd: RelativeTime
+  connectStart: RelativeTime
+  secureConnectionStart: RelativeTime
+  connectEnd: RelativeTime
+  requestStart: RelativeTime
+  responseStart: RelativeTime
+  responseEnd: RelativeTime
+  redirectStart: RelativeTime
+  redirectEnd: RelativeTime
+  decodedBodySize: number
+  traceId?: string
+}
+
+export const matchOnPerformanceObserverCallback = (
+  request: RequestCompleteEvent
+): Promise<RumPerformanceResourceTiming | undefined> => {
+  let observer: PerformanceObserver | undefined
+  return (
+    new Promise((resolve) => {
+      observer = new PerformanceObserver((list) => {
+        const entries = list.getEntries()
+        const filteredEntries = entries.filter((entry) => entry.name === request.url)
+        const candidates = filterCandidateEntries(filteredEntries, request.startClocks)
+        if (candidates.length) {
+          if (candidates.length > 2) {
+            addTelemetryDebug('more than 2 matching performance entries found')
+            resolve(undefined)
+          }
+          if (candidates.length === 2 && firstCanBeOptionRequest(candidates)) {
+            addTelemetryDebug('2 matching performance entries found')
+            resolve(candidates[1])
+          }
+          if (candidates.length === 1) resolve(candidates[0])
+        }
+      })
+      observer.observe({ entryTypes: ['resource'] })
```

Comment:
In the unlikely case that the `PerformanceObserver` callback does not fire, we do not register the fetch. 

I am unsure if there is a way around this as the fetch request may just be taking a long time, meaning setting a "timeout" would accidentally cause us to not fetch the PerformanceEntry